src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -673,6 +676,114 @@ public void testOrderOfCheckExecution() throws Exception {
         }
     }
 
+    @Test
+    public void testSkipFileOnJavaParseExceptionTrue() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+        config.addProperty("skipFileOnJavaParseException", "true");
+        config.addProperty("javaParseExceptionSeverity", "ignore");
+        config.addChild(createModuleConfig(ConstantNameCheck.class));
+
+        final File[] files = {
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException.java")),
+            new File(getPath("InputTreeWalkerProperFileExtension.java")),
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java")),
+        };
+
+        final Checker checker = createChecker(config);
+        final Map<String, List<String>> expectedViolation = new HashMap<>();
+        expectedViolation.put(getPath("InputTreeWalkerProperFileExtension.java"),
+                Collections.singletonList(
+                        "10:27: " + getCheckMessage(ConstantNameCheck.class,
+                        MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$")));
+        verify(checker, files, expectedViolation);
+    }
+
+    @Test
+    public void testSkipFileOnJavaParseExceptionFalse() throws Exception {
```

Comment:
Please put Javadoc comment to explain why this is not a regular `verifyWithInlineXmlConfig(path`

Reply:
done